### PR TITLE
chore(ci): provide squid config scripts for ci

### LINKED ISSUE #14196: All E2E test are run with a proxy running in a container
The aim of this issue is to verify that everything works as expected when the PD is opened with preconfigured proxy. 

The proxy is set ahead of running the tests, ie using a container image (`docker.io/ubuntu/squid@sha256:6a097f68bae708cedbabd6188d68c7e2e7a38cedd05a176e1cc0ba29e3bbe029`). The e2e tests are run with a custom settings parameters passed via RunnerOptions object or via System proxy.

### PR DESCRIPTION
**Pull request by odockal**
### What does this PR do?
It brings a squid configuration files and script to spin up squid proxy on GH CI runner. On e2e repository will be a workflow that will use these files.

### What issues does this PR fix or reference?
#14196 

### How to test this PR?
One can run the script on his machine which should spin up a squid proxy using podman on 127.0.0.1:3128 address. 

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: tests/playwright/scripts/setup-proxy.sh [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        echo ""
+        read -rp "  [t]eardown and recreate / [R]estart existing? (t/R): " choice
+        case "${choice}" in
+          t|T)
+            cleanup
+            ;;
+          *)
+            echo "Restarting existing container..."
+            podman restart "${PROXY_NAME}"
+            echo "Proxy restarted at 127.0.0.1:${PROXY_PORT}"
```
[the remark this answers is not in the document]
Comment (danivilla9) [reply tests/playwright/scripts/setup-proxy.sh:107]: I was able to reproduce this issue when testing the scripts. It's a bit of an edge case but I'd suggest fixing it before merging the PR

**Review by danivilla9 — APPROVED** [review]
Tested and everything worked as expected, but I think the bug flagged by coderabbit should be addressed before merging

File: tests/playwright/scripts/setup-proxy.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (odockal) [reply tests/playwright/scripts/setup-proxy.sh]: I removed whole testing of existing stuff and CI, etc. It isn ot needed, so now it is always recreate.
